src/beaconer.rs

```diff
@@ -140,16 +140,19 @@ impl Beaconer {
 
     async fn handle_received_beacon(&mut self, packet: Packet, logger: &Logger) {
         info!(logger, "received possible PoC payload: {packet:?}");
-        let report = match packet.to_witness_report() {
+        let mut report = match packet.to_witness_report() {
             Ok(report) => report,
             Err(err) => {
                 warn!(logger, "ignoring invalid witness report: {err:?}");
                 return;
             }
         };
 
+        report.pub_key = self.keypair.public_key().to_vec();
+        report.signature = report.sign(self.keypair.clone()).await.unwrap();
```

Comment:
```suggestion
        report.signature = report.sign(self.keypair.clone()).await?;
```

Reply:
the raw unwrap here will cause a panic whereas the enigmatic `?` will just throw an error up the stack

Reply:
indeed, unwraps continue to be considered evil